### PR TITLE
Add some docs to the confusing importer file

### PR DESCRIPTION
**Pull request by twymer**

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/importer/views.py [hunk 1]
```diff
@@ -29,6 +29,18 @@ def render_error(request, domain, message):
...
+
+    This post uploads the file and is when the user enters:
+
+    named_columns:
+        Whether or not the first row of the excel sheet contains
+        header strings for the columns. This defaults to True and
+        should potentially not be an option as it is always used
+        due to how important it is to see column headers
+        in the rest of the importer.
+    """
```
Comment (esoergel) [inline corehq/apps/importer/views.py]: > and is when the user enters

I think you a word or two.

File: corehq/apps/importer/views.py [hunk 2]
```diff
@@ -109,6 +121,44 @@ def excel_config(request, domain):
 @require_POST
 @require_can_edit_data
 def excel_fields(request, domain):
+    """
+    Step one of three.
```
Comment (esoergel) [inline corehq/apps/importer/views.py]: should this be "step two of three"?  `excel_config` is also listed as step one.

File: corehq/apps/importer/views.py [hunk 3]
```diff
@@ -109,6 +121,44 @@ def excel_config(request, domain):
...
+        Passed through from last step, see that for documentation
+
+    case_type:
+        The type of case we are matching to. When creating new cases,
+        this is the type they will be created as. When updating
+        existing cases, this is the type that we will search for.
+        If the wrong case type is used when looking up existing cases,
+        we will not up date them.
+
+    search_field:
```
Comment (esoergel) [inline corehq/apps/importer/views.py]: duplicated below, can probably remove

File: corehq/apps/importer/views.py [hunk 4]
```diff
@@ -109,6 +121,44 @@ def excel_config(request, domain):
...
+
+    named_columns:
+        Passed through from last step, see that for documentation
+
+    case_type:
+        The type of case we are matching to. When creating new cases,
+        this is the type they will be created as. When updating
+        existing cases, this is the type that we will search for.
+        If the wrong case type is used when looking up existing cases,
+        we will not up date them.
```
Comment (esoergel) [inline corehq/apps/importer/views.py]: trivial point, but you probably mean "update" at the end here.

File: corehq/apps/importer/views.py [hunk 5]
```diff
@@ -109,6 +121,44 @@ def excel_config(request, domain):
...
+        creating new cases only you would expect these to be blank with
+        the create_new_cases flag set.
+
+    search_field:
+        Either case id or external id, determines which type of
+        identification we are using to match to cases.
+
+    key_column/value_column:
+        These correspond to an advanced feature allowing a user
+        to modify a single case with multiple rows.
```
Comment (esoergel) [inline corehq/apps/importer/views.py]: might be a good idea to point to an upload script.  I think we've got one in our repo somewhere, don't we?  If not, I wrote one for TDH we could put in as documentation.

File: corehq/apps/importer/views.py (hunk 1 above)
Comment (twymer) [inline corehq/apps/importer/views.py]: Oh, I thought there were going to be more in the list but the string just reads strangely..

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/importer/views.py [hunk 6]
```diff
@@ -29,6 +29,18 @@ def render_error(request, domain, message):
 
 @require_can_edit_data
 def excel_config(request, domain):
+    """
```
Comment (dimagimon) [inline corehq/apps/importer/views.py]: E302 expected 2 blank lines, found 1

**Comment by twymer** [comment]
Thought to myself "Do I even need to PR docstring changes?". Apparently the answer was yes. Anyway, @esoergel I updated.

File: corehq/apps/importer/views.py (hunk 1 above)
Comment (twymer) [inline corehq/apps/importer/views.py]: The base PR review shows the old version of this docstring here, but I updated this and if you click on the file changes view you can see that.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Pull request merged**